File: backend/src/main/java/com/wodowski/backend/matching/dto/MatchingUser.java
package com.wodowski.backend.matching.dto;

import com.wodowski.backend.user.dto.ImageDTO;
import com.wodowski.backend.user.dto.Preference;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class MatchingUser {

    private String id;
    private String name;
    private String description;
    private int age;
    private boolean sex;
    private String orientation;
    private String location;
    private List<ImageDTO> photosUrls = new ArrayList<>();
    private Preference preference;
}

File: frontend/src/app/shared/models/MatchingUser.ts
import {Preference} from "./Preference";
import {File} from "./File";

export class MatchingUser{
  id: string;
  name: string;
  description: string;
  age: number;
  sex: boolean;
  orientation: string;
  location: string;
  images: File[];
  // ???
  preference: Preference;

  constructor(id: string, name: string, description: string, age: number, sex: boolean, orientation: string, location: string, images: File[], preference: Preference) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.age = age;
    this.sex = sex;
    this.orientation = orientation;
    this.location = location;
    this.images = images;
    this.preference = preference;
  }
}
